cpp/oneapi/dal/backend/primitives/distance/lp_distance_dpc.cpp

```diff
@@ -0,0 +1,62 @@
+/*******************************************************************************
+* Copyright 2021 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include "oneapi/dal/backend/primitives/distance/distance.hpp"
+
+namespace oneapi::dal::backend::primitives {
+
+template <typename Float, typename Metric>
+sycl::event distance<Float, Metric>::operator()(const ndview<Float, 2>& inp1,
+                                                const ndview<Float, 2>& inp2,
+                                                ndview<Float, 2>& out,
+                                                const event_vector& deps) const {
+    check_inputs(inp1, inp2, out);
+    // Getting raw USM pointers
+    const auto* inp1_ptr = inp1.get_data();
+    const auto* inp2_ptr = inp2.get_data();
+    auto* out_ptr = out.get_mutable_data();
+    // Getting info about dimensions
+    const auto n_features = inp1.get_dimension(1);
+    const auto n_samples1 = inp1.get_dimension(0);
+    const auto n_samples2 = inp2.get_dimension(0);
```

Comment:
It would be good to check that k1 == k2

Reply:
All checks are hidden under `check_inputs(inp1, inp2, out);`